### ISSUE TITLE
Feature Request: Blocking User

### ISSUE THREAD

**Issue by yungsenpai** [Retrospring/bugs#2]
User X Blocking user Y will lead to

* X's Profile not visible for user Y
* X's Questions and answers not visible for user Y

**Comment by pixeldesu** [comment]
Let me add something to this, because I think it would be better doing it this way.

Adding to your...
> * X's Profile not visible for user Y
> * X's Questions and answers not visible for user Y

...following should be thought about as well:

* Y's Profile not visible for user X
* Y's Questions and answers not visible for user X

This will cause that these 2 users, except if they end up making new accounts (for _stalking_), being practically invisible for each other, blocking one user basically shows you have either personal or _any_ issues with them and don't want to interact with them in any way.

**Comment by TheCoward** [comment]
I dig this. I mean, it's how Twitter works, and it works fine.

**Comment by yungsenpai** [comment]
Yeah, makes sense :)
I dont think this would be too hard to implement.

**Comment by nilsding** [comment]
@TheRealKingS told me via IRC he looks forward to implement the blocking functionalities (once he gets a local testing instance up and running, of course…)

**Comment by TheRealKingS** [comment]
Yes i'm into this. My local testing instance is running, now i will write the code. The private account feature will be added too (Maybe a new Feature Request?)

**Comment by yukimono** [comment]
I believe private accounts is being worked on too, incase you were wondering

**Comment by colons** [comment]
i am really, really, really invested in this feature existing in the, uhh, not-too-distant future

is there anything a django developer who has never touched ruby can do to help

**Comment by yukimono** [comment]
i'd tell you to try working on it in a fork, but rails has a lot to take in.

**Comment by yukimono** [comment]
still, though! if you want to try going at it, feel free to try.
it'll be a little bit before I'll work on implementing it.
(i might also have put too much on my plate at this moment)

**Comment by colons** [comment]
i might try some smaller stuff first :3

**Comment by yukimono** [comment]
offtopic:
you could try fixing bug no. 44.

**Comment by colons** [comment]
oh hey yeah i'd noticed that

there's one other, even smaller thing that's been bugging me though…

**Comment by yukimono** [comment]
offtopic:
if it's not reported yet, file it and work on it :panda_face:

**Comment by colons** [comment]
that's what im doing (/ω＼)

**Comment by colons** [comment]
btw i'd like to add that i'm way more eager to get @pixeldesu's suggestions; there are people i just don't want to see but i don't give a shit if other people see them on my questions

i suppose that's potentially a difference between 'muting' and 'blocking' though

**Comment by yukimono** [comment]
The way I outlined it in my mind.

Tom blocks communication Geoff;
Geoff cannot get notified by Tom, view any of Tom's content, or answer any questions by Tom.
Tom cannot get notified by Geoff, view any of Geoff's content, or answer any questions by Geoff.

Tom mutes Geoff;
Geoff can get notified by Tom, view Tom's content, and answer any questions by Tom.
Tom cannot get notified by Geoff, view any of Geoff's content, but can answer any questions by Geoff.

**Comment by yukimono** [comment]
I was going to make a neat little infographic for the confirmation dialog and everything

**Comment by colons** [comment]
sick as heck
